janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Text.java

```diff
@@ -120,13 +225,63 @@ public boolean isValidCondition(Object condition) {
             return condition instanceof String && StringUtils.isNotBlank(condition.toString());
         }
 
+        @Override
+        public boolean hasNegation() {
+            return true;
+        }
+
+        @Override
+        public JanusGraphPredicate negate() {
+            return Text.NOT_CONTAINS_REGEX;
+        }
+
         @Override
         public String toString() {
             return "textContainsRegex";
         }
 
     },
 
+    /**
+     * Whether the text doesnt contain a token that matches a regular expression
+     */
+    NOT_CONTAINS_REGEX {
+        @Override
+        public boolean test(Object value, Object condition) {
+            this.preevaluate(value, condition);
+            return value != null && evaluateRaw(value.toString(), (String) condition);
+        }
+
+        @Override
+        public boolean evaluateRaw(String value, String regex) {
+            for (String token : tokenize(value.toLowerCase())) {
+                if (REGEX.evaluateRaw(token,regex)) return false;
+            }
+            return true;
+        }
+
+        @Override
+        public boolean isValidCondition(Object condition) {
+            return condition instanceof String && StringUtils.isNotBlank(condition.toString());
```

Comment:
Any specific reason for `StringUtils.isNotBlank(condition.toString())`? I think blank space is also a valid regular expression.

Reply:
The existing REGEX/CONTAINS_REGEX use this test as well, so I carried it over out of consistency.

Reply:
Noted. Not sure why we did that in the first place, but it makes sense to follow the old behavior.